src/lib.rs

```diff
@@ -361,6 +362,26 @@ impl ParserConfig {
         request.parse_with_config_and_uninit_headers(buf, self, headers)
     }
 
+    /// Sets whether invalid header lines should be silently ignored in responses.
+    ///
+    /// This mimicks the behaviour of major browsers. You probably don't want this.
+    /// You should only want this if you are implementing a proxy whose main
+    /// purpose is to sit in front of browsers whose users access arbitrary content
+    /// which may be malformed, and they expect everything that works without
+    /// the proxy to keep working with the proxy.
+    ///
+    /// This option will prevent `ParserConfig::parse_response` from returning
+    /// an error encountered when parsing a header, except if the error was caused
+    /// by the character NUL (ASCII code 0), as Chrome specifically always reject
+    /// those.
+    pub fn ignore_invalid_headers_in_responses(
```

Comment:
This option equally affects requests
```suggestion
    pub fn ignore_invalid_headers(
```

Reply:
It doesn't, because we pass `&ParserConfig::default()` to `parse_headers` in `parse_request`, that's what I was talking about on Discord yesterday.

Reply:
There are new methods though like `ParserConfig::parse_request()` that allow configuring the parser for requests. Existing behavior didn't change but new programs can use this behavior.

Reply:
Yes, and as I have stated multiple times, we pass `&ParserConfig::default()` to `parse_headers` in `parse_request`. The only `parse_request` function there is in that crate is the method on `ParseConfig`.

`ParserConfig::parse_request` calls `Request::parse_with_config` which calls `Request::parse_with_config_and_uninit_headers` which calls `parse_headers_iter_uninit` with `&ParserConfig::default()`, so all those flags related to header parsing don't apply to requests. That's a feature, because we don't want to relax request parsing, just response parsing. See https://github.com/seanmonstar/httparse/pull/91#discussion_r602604778.

Reply:
Ah okay, I was missing the distinction between the config's new use in `parse_with_config_and_uninit_headers` here https://github.com/seanmonstar/httparse/blob/7fd85d187b180863013f75e38f3b4c344eafb9e4/src/lib.rs#L443-L449

And the fact that we call `parse_headers_iter_uninit` with the default config. That latter part is an oversight on my part when adding the multiple spaces flag for requests. Even though we don't currently have any flags that should override header parsing behavior in request _headers_, I think the inconsistency is surprising.

Is the argument against expanding support for `ignore_invalid_headers` to requests that your use case trusts the browser to generate valid requests? Other proxies might not be so lucky as to have that kind of constrained request input. 

I tend to agree that we shouldn't add support the new flag on requests until someone asks for it, but I think it's worth being more defensive than assuming we'll always pass `&ParserConfig::default()` to this function. It would be very easy for someone to make a subsequent change that requires customizing the config there without realizing that it would make `ignore_invalid_headers_in_responses` start applying equally to requests.

Reply:
I'll add a test that enables a response-only flag and checks that it doesn't apply to requests.